### PR TITLE
RHOAIENG-2003: feat(metadata/codeserver): update VSCode imagestream annotations with installed software

### PR DESCRIPTION
**Pull request by jiridanek**
https://issues.redhat.com/browse/RHOAIENG-2003

## Description

Took into account the good suggestions from https://issues.redhat.com/browse/RHOAIRFE-320 as well.

* The n annotations are for https://github.com/opendatahub-io/notebooks/blob/main/codeserver/ubi9-python-3.11/Pipfile
* The n-1 annotations describe https://github.com/opendatahub-io/notebooks/blob/2024a/codeserver/ubi9-python-3.9/Pipfile

I've also added the `# language=json` injection comment described at https://www.jetbrains.com/help/idea/using-language-injections.html#use-language-injection-comments. If there's something similar for people using different IDEs, we can change it for something more portable later.

Finally, I put some newlines into the annotation, to make it easier for people to read it.

Screenshot 2024-10-29 at 1 53 18 PM

## How Has This Been Tested?

I'm going to run Kustomize, copy out the rendered manifest, rename it, and put it into my cluster. Then I'll check in the dashboard how it renders.

```
oc login --web
cd manifests/base
../../kustomize-5.0.2 edit set namespace redhat-ods-application
../../kustomize-5.0.2 edit set nameprefix main-
../../kustomize-5.0.2 build . | oc apply -f -
```

Screenshot 2024-10-29 at 4 53 35 PM

Screenshot 2024-10-29 at 4 52 58 PM

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: ci/check-json.sh [hunk 1]
```diff
@@ -22,15 +34,19 @@ function check_json() {
...
+          return 1
+        fi
+
+        for json in "${tmp_dir}"/*.yml; do
+            echo "    "; cat "${json}"
+            echo -n "  > "; json_verify < "${json}" || ret_code="${?}"
+        done
     else
-	echo "    Ignoring as this file doesn't contain necessary key field '${string}' for check"
+	    echo "    Ignoring as this file doesn't contain necessary key field '${string}' for check"
```
Comment (jstourac) [inline ci/check-json.sh:49]: indentation please :slightly_smiling_face:

Comment (jiridanek) [reply ci/check-json.sh:49]: no mixing spaces and tabs next time, please ;p I was wondering what the hell is happening, how could _you_ possibly leave misaligned line in code

Comment (jiridanek) [reply ci/check-json.sh]: fixed, thanks

**Comment by jstourac** [comment]
@jiridanek thank you for this and thank you for enhancing/fixing the CI for this change.

In general this LGTM and I put just one comment regarding the addition of the UBI information here. I don't insist on the removal but I think we should consider it.

File: manifests/base/code-server-notebook-imagestream.yaml [hunk 2]
```diff
@@ -16,8 +16,26 @@ spec:
   tags:
     # N Version of the image
     - annotations:
-        opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.11"}]'
-        opendatahub.io/notebook-python-dependencies: '[{"name":"code-server","version":"4.92"}]'
+        # language=json
+        opendatahub.io/notebook-software: |
+          [
+            {"name": "UBI", "version": "9"},
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply manifests/base/code-server-notebook-imagestream.yaml]: I'm going to remove the ubi line; the consistency argument is good one, and I want to have this PR merged soon, so I'm willing to delete anything that stands in the way of that.

**Comment by atheo89** [comment]
Great representation by using the `# language=json`, imo, we should do the same to the rest of the ImageStreams to keep consistency. 
/lgtm
/approve

**Comment by jiridanek** [comment]
/override ci/prow/images

rocm failed to build

**Pull request merged**